rust/datafusion/src/physical_plan/hash_join.rs

```diff
@@ -479,37 +490,40 @@ fn build_batch(
 fn build_join_indexes(
     left: &JoinHashMap,
     right: &RecordBatch,
-    join_type: &JoinType,
+    join_type: JoinType,
     right_on: &HashSet<String>,
-) -> Result<Vec<(JoinIndex, RightIndex)>> {
+) -> Result<(UInt64Array, UInt32Array)> {
     let keys_values = right_on
         .iter()
         .map(|name| Ok(col(name).evaluate(right)?.into_array(right.num_rows())))
         .collect::<Result<Vec<_>>>()?;
 
     let mut key = Vec::with_capacity(keys_values.len());
 
+    let mut left_indices = UInt64Builder::new(0);
```

Comment:
Seems better for inner join to *not* use builder / allocate bitmap for nulls? What is the current most ergonomic way to do this?